docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,1491 @@
+However, we need the routes to be imported into the gateway router in no-overlay
+mode. We need to update the route import controller to check whether a network
+is in no-overlay mode and import the routes from the host routing table if so.
+
+#### Configurable SNAT for Egress Traffic
+
+The OVN-Kubernetes BGP feature currently disables SNAT for all pod egress
+traffic if the pod network is advertised. The only exception is for traffic
+destined to other nodes in the cluster (pod-to-other-node), to ensure nodePort
+services can be accessed across networks. This egress traffic will remain SNATed
+to the node IP before leaving the node.
+
+However, this approach assumes that all external destinations can route back to
+the pod IPs, which is often not true for destinations reached via non-BGP
+routers (like a default gateway). To give users control over pod egress traffic
+SNAT behavior, we introduce a new `outboundSNAT` field to the
+ClusterUserDefinedNetwork CRD. When enabled, pod egress traffic will be SNATed
+to the node IP before leaving the node.
+
+The `outboundSNAT` field is intended for deployments where a user wants a
+no-overlay network but has not advertised the pod network to be routable from
+external networks. When enabled, pod-to-external traffic is SNATed and forwarded
+by a default gateway, while pod-to-remote-pod traffic is not SNATed regardless
+of the value of this field.
+
+##### Shared Gateway Mode
+
+If `outboundSNAT` is enabled for a network, in the gateway router, the default
+SNAT rule will be kept for pod-to-external traffic. But pod-to-remote-pod
+traffic remains unaffected by SNAT. Therefore, the network controllers in
+ovnkube-controller will need to add an address set for the pod subnets of each
+CUDN and the default network. Then the network controllers will add an
+exemption rule with the address set to the SNAT entries.
+
+In the following example, `10.128.0.0/24` is the pod subnet in the local node,
+`10.128.0.0/16` is the CIDR of the pod network. `outboundSNAT` is enabled.
+
+```text
+# Address_Set for the no-overlay pod network CIDR
+_uuid               : a718588e-0a9b-480e-9adb-2738e524b82d
+addresses           : ["10.128.0.0/16"]
+external_ids        : {ip-family=v4}
+name                : a15397791517117706455
+
+# Address_Set for the node IPs
+_uuid               : e7906e7c-6aaa-46bf-8d26-e25cd650f59f
+addresses           : ["172.18.0.2", "172.18.0.3", "172.18.0.4"]
+external_ids        : {ip-family=v4, "k8s.ovn.org/id"="default-network-controller:EgressIP:node-ips:v4:default", "k8s.ovn.org/name"=node-ips, "k8s.ovn.org/owner-controller"=default-network-controller, "k8s.ovn.org/owner-type"=EgressIP, network=default}
+name                : a712973235162149816
+
+# per-pod SNAT rule in the Gateway Router (exclude pod-network destinations)
+_uuid               : cd9264de-f2c6-4528-8f53-c2ccb5b23c8d
+allowed_ext_ips     : []
+exempted_ext_ips    : ["a15397791517117706455"] # Exclude traffic to the pod Network
```

Comment:
I would call out we are using this to avoid the negative match that would need to happen if we used `match` instead. That's the reason right?